packages/flame_tiled/lib/src/tile_atlas.dart

```diff
@@ -100,7 +100,7 @@ class TiledAtlas {
       // really boring map.
       final image = (await Flame.images.load(key)).clone();
 
-      return atlasMap[key] ??= TiledAtlas._(
+      return atlasMap[key] = TiledAtlas._(
```

Comment:
A question, if the atlas keys are same for 2 maps, wouldn't that case be handled by existing `atlasMap.containsKey(key)` few lines above?

Reply:
You are absolutely right, my test case was terrible. Thanks for pointing it out!

Also, my explanation of this issue also has some flaws.
As you seen, maps are handled by `if(atlasMap.containsKey(key))` if they are loaded sequentially😂 

But if they are loaded in the race condition, both of them would be stopped at: 
`final image = (await Flame.images.load(key)).clone();`.
Then they keep going...
As a result, they reach `??=` without `if(atlasMap.containsKey(key))` check.

So the test case should be changed from:
```dart
final component1 = await TiledComponent.load(
  'single_tile_map_1.tmx',
  Vector2(16, 16),
);
final component2 = await TiledComponent.load(
  'single_tile_map_2.tmx',
  Vector2(16, 16),
);
```
to: 
```dart
final components = await Future.wait([
  TiledComponent.load(
    'single_tile_map_1.tmx',
    Vector2(16, 16),
  ),
  TiledComponent.load(
    'single_tile_map_2.tmx',
    Vector2(16, 16),
  )
]);
```
The difference between these is racing or not.

**But even after changing the test case, it passes with and without this fix.**

This is because of here:
https://github.com/flame-engine/flame/blob/f125593aaaaef160395b772180b1514f6be3ac4f/packages/flame_tiled/lib/src/renderable_layers/renderable_layer.dart#L46
This line clones `TiledAtlas` one more time. So it was safe indeed.

You can remove this `clone()` and then see the (updated) test would be failed with `??=`.

But still, I think it should be changed to `??=` to `=` because of 
1) Consistency with others like more than one tileset
2) It's safe to use without a thorough understanding of this class (maybe extended from 1 )


> Probably, you have been curious about why I had an issue without any problem in the code.
That is because I am using the forked flame version with optimized TiledLayer without `clone()` one more time.
But regardless of my forked version, this would be better to be fixed.
I am thoroughly testing my patch in production and will show up to you guys

Reply:
Okay, that makes sense. So in that case, does something following be better?
```suggestion
      return atlasMap.contains(key) ? atlasMap[key]!.clone() : atlasMap[key] = TiledAtlas._(
```
It is a bit verbose, but it avoids replacing the atlas in the map if it is already loaded by some other async call.

In any case, consider adding at least a comment at the return about this special case 🙂.

Reply:
Sounds great!

But this is another option.
```dart
if (atlasMap.containsKey(key)) {
  return atlasMap[key]!.clone();
}

return TiledAtlas._(
  atlas: image,
  offsets: {key: Offset.zero},
  key: key,
);
```
Which one do you prefer? I will follow your decision.

Reply:
Go ahead with your option. It looks more readable👍🏼

Reply:
Final option:
just overwrite atlasMap[key] without checking if exists.
IMHO, it has less code but leads to the same result.

Reply:
From the looks of it, the whole point of having `atlasMap` seems to be to avoid atlas recreations. But even after caching, we are anyways returning a new or cloned version. So I guess even overwriting without checking should technically be fine. Also, this part of code is not in critical path, so I'd say prefer whichever option is more readable.🙂